python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -329,6 +333,36 @@ public void visitAssignmentStatement(AssignmentStatement assignmentStatement) {
       });
   }
 
+  @Override
+  public void visitParameter(Parameter parameter) {
+    scan(parameter.typeAnnotation());
+    scan(parameter.defaultValue());
+    Optional.ofNullable(parameter.typeAnnotation())
+      .map(TrivialTypeInferenceVisitor::resolveTypeAnnotationType)
+      .ifPresent(type -> setTypeToName(parameter.name(), type));
+    scan(parameter.name());
+  }
+
+  private static PythonType resolveTypeAnnotationType(TypeAnnotation typeAnnotation) {
+    if (typeAnnotation.expression() instanceof Name name && name.typeV2() != PythonType.UNKNOWN) {
+      return new ObjectType(name.typeV2(), TypeSource.TYPE_HINT);
```

Comment:
Should we also guard against having nested `ObjectType` here somehow?
What will we have if the annotation is a type alias? (This case, I feel, can be left out of scope, but it could be interesting to have a test for it nevertheless).

Reply:
I didn't find a Python code that could represent it

Reply:
Fair enough, and it most likely makes sense not to have such check everywhere anyway.
I'm still concerned that I feel it's theoretically possible to create such nested structures, but we can handle that in another PR/ticket.